oteapi_optimade/dlite/parse.py

```diff
@@ -0,0 +1,280 @@
+"""OTEAPI strategy for parsing OPTIMADE structure resources to DLite instances."""
+# pylint: disable=invalid-name,too-many-branches,too-many-statements
+import logging
+import sys
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+import dlite
+from optimade.adapters import Structure
+from optimade.models import StructureResponseMany, StructureResponseOne, Success
+from oteapi.models import SessionUpdate
+from oteapi_dlite.models import DLiteSessionUpdate
+from oteapi_dlite.utils import get_collection, update_collection
+from pydantic import BaseModel, ValidationError
+from pydantic.dataclasses import dataclass
+
+from oteapi_optimade.exceptions import OPTIMADEParseError
+from oteapi_optimade.models import OPTIMADEDLiteParseConfig, OPTIMADEParseSession
+from oteapi_optimade.strategies.parse import OPTIMADEParseStrategy
+
+if TYPE_CHECKING:  # pragma: no cover
+    from typing import Any, Dict, Optional, Union
+
+
+LOGGER = logging.getLogger("oteapi_optimade.dlite")
+LOGGER.setLevel(logging.DEBUG)
+LOGGER.addHandler(logging.StreamHandler(sys.stdout))
+
+
+@dataclass
+class OPTIMADEDLiteParseStrategy:
+    """Parse strategy for JSON.
+
+    **Implements strategies**:
+
+    - `("mediaType", "application/vnd.optimade+dlite")`
```

Comment:
Do we really want to spell the same thing in so many ways? What about standardising on lower case? Other capitalisations could easily be supported with a simple tolower().

Reply:
[RFC 2045](https://www.rfc-editor.org/rfc/rfc2045) states that types and subtypes are not case sensitive. Because of this we should consider updating the OTEAPI Core factory to reflect this. Suggest to leave this as-is for now, and make a follow-up task to simplify the strategy declaration

Reply:
I've opened the issue EMMC-ASBL/oteapi-core#253 to follow up on this.